hail/src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -182,15 +183,40 @@ class StagedRegionValueBuilder private(val mb: MethodBuilder, val typ: PType, va
     }
   }
 
-  def addBoolean(v: Code[Boolean]): Code[Unit] = region.storeByte(currentOffset, v.toI.toB)
+  def checkType(knownType: Type): Unit = {
+    val current = ftype match {
```

Comment:
The java test that's currently failing is failing due to this match. It is possible for `ftype` to be a type other than these two values.

Reply:
Er, maybe this should be fundamental type. I'll fix the test failures.